### PR TITLE
Update typings for BeforeRequestHook

### PR DESCRIPTION
**Pull request by deini**
`BeforeRequestHook` types got out of date.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ export type Input = Request | URL | string;
 export type BeforeRequestHook = (
 	input: Input,
 	options: NormalizedOptions,
-) => void | Promise<void>;
+) => Response | void | Promise<void>;
```
Comment (sholladay) [inline index.d.ts]: We also need `Promise<Response>`, if I'm not mistaken. Would be good to add a test case for it in the type tests, too.

Comment (deini) [reply index.d.ts]: Good catch, updating.

File: index.test-d.ts [hunk 2]
```diff
@@ -37,8 +37,8 @@ ky(url, {
 			(_input, _options) => {
 				return new Response('Test');
 			},
-			(_input, _options) => {
-				return Promise.resolve(new Response('Test'));
+			async (_input, _options) => {
```
Comment (deini) [inline index.test-d.ts]: 💯

**Comment by sholladay** [comment]
Huh yet another strange error in CI that only occurs on Node 12 and isn't code related...

```js
Error {
  message: 'Protocol error (Target.setDiscoverTargets): Target closed.',
}

node_modules/puppeteer/lib/Connection.js:74:56
Connection.send (node_modules/puppeteer/lib/Connection.js:73:12)
Function.create (node_modules/puppeteer/lib/Browser.js:34:22)
Launcher.launch (node_modules/puppeteer/lib/Launcher.js:184:37)
withPage (test/helpers/with-page.js:4:18)
```

**Comment by sholladay** [comment]
Got CI to pass by re-running it twice.

Thanks for the PR, @deini!

**Pull request merged**